pkg/gitparse/gitparse.go

```diff
@@ -157,6 +164,8 @@ type Commit struct {
 	Date    time.Time
 	Message strings.Builder
 	Size    int // in bytes
+
+	hasDiffs bool
```

Comment:
It doesn't look like this _needs_ to be a struct field - it looks like it's got a lifetime that's compatible with just being a `FromReader` local variable. Did you consider tracking it there instead of here? I can see arguments both ways.

Reply:
I felt encapsulating it within the `Commit` struct made it easier to reason about. The `FromReader` method as it is has a lot going on and adding a another variable was less easier to reason about. Happy to make it a local var if you feel strongly about it 😄